mozilla_vpn/views/subscriptions.view.lkml

```diff
@@ -252,6 +291,130 @@ view: subscriptions__active {
   }
 }
 
+view: original_subscriptions__active {
+  dimension_group: active {
+    type: time
+    timeframes: [
+      raw,
+      date,
+      week,
+      month,
+      quarter,
+      year
+    ]
+    sql: ${TABLE};;
+    convert_tz: no
+    datatype: date
+  }
+
+  dimension: is_end_of_month {
+    type: yesno
+    sql: ${active_raw} = LAST_DAY(${active_raw}, MONTH) OR ${active_raw} = DATE(${metadata.last_modified_date}) - 1;;
+  }
+
+  dimension: is_end_of_quarter {
+    type: yesno
+    sql: ${active_raw} = LAST_DAY(${active_raw}, QUARTER) OR ${active_raw} = DATE(${metadata.last_modified_date}) - 1;;
+  }
+
+  dimension: is_end_of_year {
+    type: yesno
+    sql: ${active_raw} = LAST_DAY(${active_raw}, YEAR) OR ${active_raw} = DATE(${metadata.last_modified_date}) - 1;;
+  }
+
+  dimension: max_active_date {
+    description: "Get max active date from end date in active date filter.  If null, use last modified date."
+    hidden: yes
+    type: date
+    sql: COALESCE({% date_end active_date %}, ${metadata.last_modified_date})-1;;
+    # Using date_start and date_end with date filters; it is NULL if active_date is not specified or open-ended.
+  }
+
+  dimension: is_max_active_date {
+    type: yesno
+    sql:  ${active_raw}=${max_active_date};;
+  }
+}
+
+
+view: original_subscriptions__retention {
+  dimension: months_since_original_subscription_start {
+    description: "Months since original subscription start date"
+    type: number
+    sql: ${TABLE};;
+  }
+
+  # dimension: original_subscription_id {
+  #   primary_key: yes
+  #   hidden: yes
+  #   sql: COALESCE(${subscriptions.original_subscription_id}, ${subscriptions.subscription_id});;
+  # }
+
+  dimension: primary_key {
+    primary_key: yes
+    hidden: yes
+    # sql: COALESCE(${subscriptions.original_subscription_id},${subscriptions.subscription_id} )|| ${months_since_original_subscription_start};;
+    sql: COALESCE(${subscriptions.original_subscription_id},${subscriptions.subscription_id} );;
```

Comment:
I see! It makes sense that the primary key should be unique per each row but do we always need to set a primary key when it is not ended up being used?
(This question is a general question not an objection to your suggestion which I agree to change. )

Reply:
You don't necessarily always need to set up a primary key dimension (i.e. with `primary_key: yes`) but if you do it's important that it be correct as it will be used by measures with `type: count`.

I think the main benefit of defining a primary key dimension is to allow Looker to calculate [symmetric aggregates](https://cloud.google.com/looker/docs/best-practices/understanding-symmetric-aggregates).